MarvelApiClient/CharactersAPIClient.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  CharactersAPIClient.swift
+//  MarvelAPIClient
+//
+//  Created by Pedro Vicente on 14/11/15.
+//  Copyright © 2015 GoKarumi S.L. All rights reserved.
+//
+
+import Foundation
+import BothamNetworking
+import Result
+
+public class CharactersAPIClient {
+
+    private let apiClient: BothamAPIClient
+    private let parser: CharactersParser
+
+    init(apiClient: BothamAPIClient, parser: CharactersParser) {
+        self.parser = parser
+        self.apiClient = apiClient
+    }
+
+    public func getAll(offset offset: Int, limit: Int,
+        completion: (Result<GetCharactersDTO, BothamAPIClientError>) -> ()) {
+        assert(offset >= 0 && limit >= 0)
+        let params: [String:String?] =  [MarvelAPIParams.offset : "\(offset)", MarvelAPIParams.limit : "\(limit)"]
```

Comment:
I remember why we changed this. This is the ``Alamofire`` implementation:

```swift
// MARK: - Request

    /**
        Creates a request for the specified method, URL string, parameters, parameter encoding and headers.
        - parameter method:     The HTTP method.
        - parameter URLString:  The URL string.
        - parameter parameters: The parameters. `nil` by default.
        - parameter encoding:   The parameter encoding. `.URL` by default.
        - parameter headers:    The HTTP headers. `nil` by default.
        - returns: The created request.
    */
    public func request(
        method: Method,
        _ URLString: URLStringConvertible,
        parameters: [String: AnyObject]? = nil,
        encoding: ParameterEncoding = .URL,
        headers: [String: String]? = nil)
        -> Request
    {
        let mutableURLRequest = URLRequest(method, URLString, headers: headers)
        let encodedURLRequest = encoding.encode(mutableURLRequest, parameters: parameters).0
        return request(encodedURLRequest)
    }
```

Right now I'd like to change the API to use [String:String] instead of [String:String?] but this is just my opinion and the code is working. This is something we can discuss with the rest of the team.